src/lib/gossip_net/libp2p.ml

```diff
@@ -470,8 +471,8 @@ module Make (Rpc_intf : Mina_base.Rpc_intf.Rpc_interface_intf) :
                     in
                     upon (after restart_after) (fun () ->
                         don't_wait_for
-                          (let%bind () = Mina_net2.shutdown n in
-                           on_unexpected_termination ()))
+                          (let%map () = Mina_net2.shutdown n in
```

Comment:
I changed `on_unexpected_termination : unit -> unit Deferred.t` into `restart_libp2p : unit -> unit` because the name was clearer and the `Deferred.t` was not actually used. I changed `bind` to `map` to accommodate for this change to the return type.